Source/WebCore/html/URLRegistry.h

```diff
@@ -46,9 +47,14 @@ class URLRegistrable {
 class URLRegistry {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    virtual ~URLRegistry() = default;
-    virtual void registerURL(ScriptExecutionContext&, const URL&, URLRegistrable&) = 0;
-    virtual void unregisterURL(const URL&) = 0;
+    static Vector<URLRegistry*> allRegistries();
```

Comment:
I probably would have had this return something that iterates across references rather than a vector of pointers, just to make the caller code look nicer. But I suppose that would still require a `Vector` underneath it, or maybe a `FixedVector` since we never need to modify it. Another alternative is to just expose a `forEach` function instead.

Reply:
Ok, I'll look into a forEach() function. I hesitated going this way. I eventually went with the Vector because it allowed the call site to easily stop iterating when finding the right registry. That said, there are only 2 registries at the moment so not an important optimization.